vmm/src/api/http.rs

```diff
@@ -305,7 +306,9 @@ pub fn start_http_path_thread(
 ) -> Result<thread::JoinHandle<Result<()>>> {
     std::fs::remove_file(path).unwrap_or_default();
     let socket_path = PathBuf::from(path);
-    let server = HttpServer::new(socket_path).map_err(Error::CreateApiServer)?;
```

Comment:
Nit: With this refactoring, `HttpServer::new()` will only be used in its own unit tests and could be removed in the future.